src/kitovu/sync/syncing.py

```diff
@@ -12,34 +12,36 @@
 from kitovu.sync.plugin import smb
 
 
-def _find_plugin(pluginname: str) -> syncplugin.AbstractSyncPlugin:
+def _find_plugin(pluginname: str,
+                 reporter: utils.AbstractReporter) -> syncplugin.AbstractSyncPlugin:
     """Find an appropriate sync plugin with the given settings."""
     builtin_plugins = {
-        'smb': smb.SmbPlugin(),
+        'smb': smb.SmbPlugin(reporter),
     }
     if pluginname in builtin_plugins:
         return builtin_plugins[pluginname]
 
     try:
         manager = stevedore.driver.DriverManager(namespace='kitovu.sync.plugin',
-                                                 name=pluginname, invoke_on_load=True)
+                                                 name=pluginname, invoke_on_load=True,
+                                                 invoke_args=(reporter))
```

Comment:
You just found another things that sucks in Python: The syntax for single-element tuples... `(reporter)` is not a tuple, it's the same thing as just `reporter` (or e.g. `((reporter))`).

A single-element tuple is `(reporter, )` (because it's actually the comma creating the tuple syntactically... except for `()` which is special cased to be an empty tuple).

Yes, it sucks :laughing:. I'll fix it up while merging, though.

Reply:
Oh yeah, that's unexpected... Thanks for fixing it 👍